types/http-assert/http-assert-tests.ts

```diff
@@ -1,26 +1,115 @@
-import httpAssert = require('http-assert');
-import { HttpError } from 'http-errors';
+import httpAssert = require("http-assert");
+
+declare function unknown(): unknown;
+
+// assert()
+
+(() => {
+    const value = unknown();
+    httpAssert(typeof value === "string");
+
+    // $ExpectType string
+    value;
+})();
+
+(() => {
+    const value = unknown() as string | undefined;
+    httpAssert(value);
+
+    // $ExpectType string
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert(typeof value === "number", 400);
+
+    // $ExpectType number
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert(typeof value === "boolean", 400, "message");
+
+    // $ExpectType boolean
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert(value === undefined, 400, { foo: "bar" });
+
+    // $ExpectType undefined
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert(value === null, 400, "message", { foo: "bar" });
+
+    // $ExpectType null
+    value;
+})();
+
+// assert.ok()
+
+(() => {
+    const value = unknown();
+    httpAssert.ok(typeof value === "string");
+
+    // $ExpectType string
+    value;
+})();
+
+(() => {
+    const value = unknown() as string | undefined;
+    httpAssert.ok(value);
+
+    // $ExpectType string
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert.ok(typeof value === "number", 400);
+
+    // $ExpectType number
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert.ok(typeof value === "boolean", 400, "message");
+
+    // $ExpectType boolean
+    value;
+})();
+
+(() => {
+    const value = unknown();
+    httpAssert.ok(value === null, 400, "message", { foo: "bar" });
+
+    // $ExpectType null
+    value;
+})();
+
+// Other assertions
 
 const status = 401;
-const message = 'some error message';
+const message = "some error message";
```

Comment:
Can you please revert changing `'` to `"` so the diff is more sensible?

Reply:
I reformatted the file to use single quotes, as per the Prettier configuration as this was basically the only line that stayed the same.

The original file was moved into the ts3.6 folder and a new file with the same name was created where coincidentally some lines are the same.